Along with minor enhancements, added Search Results Export API, Single Study Detail API, Trial Timelines API and Trial Dashboard API. Updated backend/README.md and View definition.

bragu opened the pull request:
Following changes are covered in this PR.

1. Countries API - Blank value is removed and US is at the top.
2. Created API for conditions ‘auto-suggest’
3. Search Studies API now handle multi-select parameters
4. Added parameters additional condition terms, state and  city to Search Studies API and modified view accordingly
5. Created API for City and State list
6. Created API to get single study details based on nct_id
7. Created Trial Timelines API
8. Created Trials Dashboard API to return metadata required for visualization
9. Created Search Results Export API

**Recreate database view to take care of additional parameters.
1. docker exec -it --user postgres pgdb /bin/bash
2. psql -d aact -f database/scripts/search_studies.sql**

bragu committed:
Added Search Results Export API, Single Study Detail API, Trial Timelines API and Trial Dashboard API. Updated backend/README.md accordingly.

bragu committed:
Updated view definition in database/scripts/search_studies.sql to account for additional parameters city and state

bragu committed:
Merge branch 'main' into feature/api

swt144 commented on backend/ctgov/api/views.py:
```diff
@@ -69,240 +82,219 @@ def get(self, request, *args, **kwargs):
+            .exclude(country="United States")
             .order_by("country")
+        )
+        qs2 = (
+            Facilities.objects.all()
+            .distinct()
+            .filter(country="United States")
```
Since there are performance issues on the AWS hosted site, I would highly recommend hard coding this value instead of an additional database pull for a known value.

bragu commented:
Merged PR# 56 (Along with minor enhancements, added Search Results Export API, Single Study Detail API, Trial Timelines API and Trial Dashboard API. Updated backend/README.md and View definition.)
